### PR TITLE
Refresh plugin for June 2023

### PR DESCRIPTION
**Pull request by basil**
### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <extension>
     <groupId>io.jenkins.tools.incrementals</groupId>
     <artifactId>git-changelist-maven-extension</artifactId>
-    <version>1.3</version>
+    <version>1.6</version>
```
Comment (basil) [inline .mvn/extensions.xml:5]: Latest version at the time of this writing.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,5 +1,4 @@
-buildPlugin(configurations: [
-  [ platform: "linux", jdk: "8", jenkins: null ],
-  [ platform: "linux", jdk: "11", jenkins: null ],
-  [ platform: "windows", jdk: "8", jenkins: null ]
+buildPlugin(useContainerAgent: true, configurations: [
+  [platform: 'linux', jdk: 17],
+  [platform: 'windows', jdk: 11],
```
Comment (basil) [inline Jenkinsfile:3]: Following the recommendation of the archetype.

File: pom.xml [hunk 3]
```diff
@@ -39,7 +39,6 @@ limitations under the License.
   <properties>
     <revision>2.32</revision>
     <changelist>-SNAPSHOT</changelist>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (basil) [inline pom.xml:42]: Already in the plugin parent POM.

File: pom.xml [hunk 4]
```diff
@@ -59,8 +58,7 @@ limitations under the License.
     <jclouds.version>2.5.0</jclouds.version>
     <!-- Depend on latest Jenkins 2.x LTS
          See README of https://github.com/jenkinsci/plugin-pom/ -->
-    <jenkins.version>2.332.1</jenkins.version>
-    <java.level>8</java.level>
```
Comment (basil) [inline pom.xml:63]: As in the archetype.

File: pom.xml [hunk 5]
```diff
@@ -106,16 +104,16 @@ limitations under the License.
...
+        <version>2102.v854b_fec19c92</version>
         <scope>import</scope>
         <type>pom</type>
       </dependency>
     </dependencies>
   </dependencyManagement>
 
   <scm>
-    <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
```
Comment (basil) [inline pom.xml:116]: GitHub no longer supports the `git://` protocol.

File: pom.xml [hunk 6]
```diff
@@ -169,7 +167,6 @@ limitations under the License.
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>config-file-provider</artifactId>
-      <version>3.7.1</version>
```
Comment (basil) [inline pom.xml:172]: Using the version managed in the Jenkins plugin BOM.

File: pom.xml [hunk 7]
```diff
@@ -411,23 +408,6 @@ limitations under the License.
...
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-javadoc-plugin</artifactId>
-            <configuration>
-              <additionalparam>-Xdoclint:none</additionalparam>
-            </configuration>
-          </plugin>
-        </plugins>
-      </build>
-    </profile>
```
Comment (basil) [inline pom.xml:430]: Jenkins no longer supports Java 8.

**Comment by felfert** [comment]
LGTM

**Review by felfert — APPROVED** [review]
LGTM

**Pull request merged**